pepagent/pepagent.py

```diff
@@ -66,222 +76,362 @@ def close_connection(self) -> None:
+        :param tag: tag (or version) of the project
+        :param anno: dict with annotations about current project
+        :param update: boolean value if project hase to be updated
+        """
+        cursor = self.postgresConnection.cursor()
+
+        proj_dict = project.to_dict(extended=True)
+        if name:
+            proj_name = name
+        else:
+            proj_name = proj_dict["name"]
+
+        proj_digest = self._create_digest(proj_dict)
+
+        # adding project status to db:
+        if STATUS_KEY in anno:
+            proj_status = anno[STATUS_KEY]
+            del anno[STATUS_KEY]
+        else:
+            proj_status = DEFAULT_STATUS
+
+        anno_info = {
+            "proj_description": proj_dict["description"],
+            "n_samples": len(project.samples),
+            "last_update": str(datetime.datetime.now()),
+            "status": proj_status,
+        }
+
+        if anno:
+            anno_info.update(anno)
+        anno_info = json.dumps(anno_info)
+        proj_dict = json.dumps(proj_dict)
+
+        if self.project_exists(namespace=namespace, name=proj_name, tag=tag):
+            try:
+                _LOGGER.info(f"Updating {proj_name} project...")
+                sql = f"""UPDATE {DB_TABLE_NAME}
+                    SET {DIGEST_COL} = %s, {PROJ_COL}= %s, {ANNO_COL}= %s
+                    WHERE {NAMESPACE_COL} = %s and {NAME_COL} = %s and {TAG_COL} = %s;"""
+                cursor.execute(
+                    sql,
+                    (
+                        proj_digest,
+                        proj_dict,
+                        anno_info,
+                        namespace,
+                        proj_name,
+                        tag,
+                    ),
+                )
+                _LOGGER.info(
+                    f"Project '{namespace}/{proj_name}:{tag}' has been updated!"
+                )
+            except psycopg2.Error:
+                _LOGGER.error("Error occurred while updating the project!")
+        else:
+            _LOGGER.error("Project does not exist! No project will be updated!")
+
+    def get_project_by_registry(self, registry_path: str = None):
+        """
+        Retrieving project from database by specifying project registry_path
+        :param registry_path: project registry_path [e.g. namespace/name:tag]
+        :return: peppy object with found project
+        """
+        if not registry_path:
+            _LOGGER.error("No registry path provided! Returning empty project!")
+            return peppy.Project()
+        else:
+            reg = ubiquerg.parse_registry_path(registry_path)
+            namespace = reg["namespace"]
+            name = reg["item"]
+            tag = reg["tag"]
+        return self.get_project(namespace=namespace, name=name, tag=tag)
+
     def get_project(
         self,
-        registry: str = None,
-        namespace: str = None,
+        *,
```

Comment:
It might be the way the diff is rendering on my browser, but what is this `*` argument?

Reply:
We shouldn't use it here. Fixed!